config/v1/types_infrastructure.go

```diff
@@ -22,12 +23,87 @@ type Infrastructure struct {
 
 // InfrastructureSpec contains settings that apply to the cluster infrastructure.
 type InfrastructureSpec struct {
-	// cloudConfig is a reference to a ConfigMap containing the cloud provider configuration file.
-	// This configuration file is used to configure the Kubernetes cloud provider integration
-	// when using the built-in cloud provider integration or the external cloud controller manager.
-	// The namespace for this config map is openshift-config.
+	// VSphereInfrastructureConfig specifies configuration when the cluster is installed on a VSphere platform.
+	// This is only settable if the platform type is VSphere.
 	// +optional
-	CloudConfig ConfigMapFileReference `json:"cloudConfig"`
+	VSphereInfrastructureConfig *VSphereInfrastructureConfig `json:"vsphereInfrastructureConfig,omitempty"`
+}
+
+// VSphereInfrastructureConfig specifies the configuration of one or more Virtual Centers where
+// the control plane and worker nodes are running in the cluster.  This information is used to
+// control the configuration of the vsphere cloud provider support in Kubernetes as supported
+// by OpenShift.
+type VSphereInfrastructureConfig struct {
+	// secretRef is name of the authentication secret for access to the VSphere infrastructure.
+	SecretRef corev1.SecretReference `json:"secretRef"`
+
+	// port is the vCenter Server Port.
+	// defaults to 443 if not specified.
+	// +optional
+	Port *int32 `json:"port,omitempty"`
+
+	// insecure if true means vCenter is configured with a self-signed cert.
+	// defaults to false.
+	// +optional
+	Insecure bool `json:"insecure,omitempty"`
```

Comment:
how many users you do think have `insecure`. Would it be better to always require secure ?

Reply:
A lot of them run with `insecure` default unfortunately.. :(

Reply:
again, i do not think we can require a change to infrastructure.